tests/e2e/pact/broker/verify.ts

```diff
@@ -1,43 +1,86 @@
+      providerVersion: process.env.PACT_VERSION,
+      providerBaseUrl: process.env.PACT_PROVIDER_URL,
+      logLevel: "info",
+      stateHandlers: {
+        "is authenticated": () => {
+          token = `${process.env.APIGEE_ACCESS_TOKEN}`
+          Promise.resolve(`Valid bearer token generated`)
+        },
+        "is not authenticated": () => {
+          token = ""
+          Promise.resolve(`Invalid bearer token generated`)
+        }
+      },
+      requestFilter: (req) => {
+        req.headers["x-smoke-test"] = "1"
+        req.headers["Authorization"] = `Bearer ${token}`
+        return req
+      },
+      pactUrls: isLocal 
+        ? [
+          `${process.cwd()}/pact/pacts/${process.env.PACT_CONSUMER}+${endpoint}+${process.env.PACT_VERSION}-${process.env.PACT_PROVIDER}+${process.env.PACT_VERSION}.json`
+        ]
+        : []
+    })
+    return await verifier.verifyProvider()
+}
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+async function verifyConvert(): Promise<any> {
+  endpoint = "convert"
+  sleepMs = 0
+  await verify().catch(verify).catch(verify)
+}
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+async function verifyPrepare(): Promise<any> { 
+  endpoint = "prepare"
+  sleepMs = 0
+  await verify().catch(verify).catch(verify)
+}
 
 /* eslint-disable  @typescript-eslint/no-explicit-any */
-async function verify(): Promise<any> {
-  const isLocal = process.env.PACT_PROVIDER_URL === "http://localhost:9000"
-  const verifier =  new Verifier({
-    publishVerificationResult: !isLocal,
-    pactBrokerUrl: isLocal ? undefined : process.env.PACT_BROKER_URL,
-    pactBrokerUsername: process.env.PACT_BROKER_BASIC_AUTH_USERNAME,
-    pactBrokerPassword: process.env.PACT_BROKER_BASIC_AUTH_PASSWORD,
-    consumerVersionSelectors: [
-      {
-        pacticipant: `${process.env.PACT_CONSUMER}+${process.env.PACT_VERSION}`,
-        version: process.env.PACT_VERSION,
-        latest: false,
-        all: false
-      }
-    ],
-    provider: `${process.env.PACT_PROVIDER}+${process.env.PACT_VERSION}`,
-    providerVersion: process.env.PACT_VERSION,
-    providerBaseUrl: process.env.PACT_PROVIDER_URL,
-    logLevel: isLocal? "debug" : "info",
-    customProviderHeaders: [
-      "x-smoke-test: 1"
-    ],
-    requestFilter: (req, res, next) => {
-      req.headers['Authorization'] = req.headers['Authorization'] ?? `Bearer ${process.env.APIGEE_ACCESS_TOKEN}`
-      next()
-    },
-    pactUrls: isLocal 
-      ? [
-        `${process.cwd()}//pact/pacts/${process.env.PACT_CONSUMER}+${process.env.PACT_VERSION}-${process.env.PACT_PROVIDER}+${process.env.PACT_VERSION}.json`
-      ]
-      : []
-  })
-  
-  return await verifier.verifyProvider()
+async function verifyProcess(): Promise<any> { 
+  endpoint = "process"
+  sleepMs = 0
+  await verify().catch(verify).catch(verify)
 }
 
-(async () => {
-  verify().catch(verify).catch(verify)
+(async () => {  
```

Comment:
Similar thing here, would it be useful to add a catch for process, and add some logging to each catch? Or would that extra logging not add anything?

Reply:
Logging will still come from the pact binary, have found catching an error and logging it here just duplicates whatevers already been pushed to stdout from the pact binary

Reply:
:+1: